service/history/api/respondworkflowtaskcompleted/workflow_task_completed_handler.go

```diff
@@ -103,9 +103,9 @@ type (
 	}
 
 	workflowTaskFailedCause struct {
-		failedCause     enumspb.WorkflowTaskFailedCause
-		causeErr        error
-		workflowFailure *failurepb.Failure
+		failedCause               enumspb.WorkflowTaskFailedCause
+		causeErr                  error
+		workflowTerminationReason *failurepb.Failure
```

Comment:
Since terminate only supports a reason, you should refactor and only accept a string.